python/cudf/cudf/core/dataframe.py

```diff
@@ -1521,6 +1521,35 @@ def reset_index(self, drop=False, inplace=False):
             return out.set_index(RangeIndex(len(self)))
 
     def take(self, positions):
```

Comment:
Can we just move this whole function to `Frame`? There's implementations for:

- DataFrame: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.take.html
- Series: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.take.html
- Index: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.take.html
- MultiIndex: There isn't one, but there probably should be...

Reply:
Yes - I'll do that in this PR